core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -1385,10 +1391,34 @@ enum SafeRexVisitor implements RexVisitor<Boolean> {
     }
 
     @Override public Boolean visitCall(RexCall call) {
-      if (!safeOps.contains(call.getKind())) {
+      SqlKind sqlKind = call.getKind();
+      SqlOperator sqlOperator = call.getOperator();
+
+      switch (sqlKind) {
+      case DIVIDE:
```

Comment:
these features are really not related to the issue being solved.
if you want to improve the precision of the safety computation, I think you should file a separate issue and move these changes there.
we should also discuss the APIs for doing that

Reply:
Provide custom interfaces to implement different rules?